<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="use_sim_time"            default="true" />
  <arg name="param_file"     value="$(find nvblox_ros)/config/min_parameters.yaml"/>

  <node pkg="nvblox_ros" type="nvblox_node" name="nvblox_node" respawn="false" output="screen" clear_params="true">
    <rosparam file="$(arg param_file)"/>
    <remap from="depth/image" to="/depth_camera_rear_upper/aligned_depth_to_color/image_raw" />
    <remap from="depth/camera_info" to="/depth_camera_rear_upper/aligned_depth_to_color/camera_info" />

    <remap from="color/image" to="/depth_camera_rear_upper/color/image_raw" />
    <remap from="color/camera_info" to="/depth_camera_rear_upper/color/camera_info" />

    <remap from="pointcloud" to="/lidar/point_cloud" />

    <!-- Anything here ovewrites the parameters in the config file. -->
    <param name="global_frame" value="map" />
    <param name="use_sim_time" value="$(arg use_sim_time)"/>
    <param name="lidar_width" value="440" />
    <param name="use_depth" value="true" />
    <param name="use_lidar" value="true" />
    <param name="voxel_size" value="0.02" />
  </node>

</launch>
